File: Python3/3063. Linked List Frequency.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def frequenciesOfElements(self, head: Optional[ListNode]) -> Optional[ListNode]:
        freqs = collections.defaultdict(int)
        while head:
            freqs[head.val] += 1
            head = head.next

        dummy = ListNode(0)
        cur = dummy

        for v in freqs.values():
            cur.next = ListNode(v)
            cur = cur.next

        return dummy.next

File: Java/3063. Linked List Frequency.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode frequenciesOfElements(ListNode head) {
        Map<Integer, Integer> freqs = new HashMap<>();
        while (head != null) {
            freqs.put(head.val, freqs.getOrDefault(head.val, 0) + 1);
            head = head.next;
        }

        ListNode dummy = new ListNode();
        ListNode cur = dummy;
        for (int v: freqs.values()) {
            cur.next = new ListNode(v);
            cur = cur.next;
        }

        return dummy.next;
    }
}